core/network.rb

```diff
@@ -109,7 +76,7 @@ def self.getKeyFile(name)
       else
         mdbci_node = $session.mdbciNodes.find { |elem| elem[0].to_s == node_arg }
         raise "MDBCI node is not found in #{dir}" if mdbci_node.nil?
-        key_path = getBoxParameterKeyPath(dir,mdbci_node,pwd)
+        key_path = getBoxParameterKeyPath(dir.path,mdbci_node,pwd)
```

Comment:
This change may not be more relevant as you now return the directory as a result of the `extract_directory_and_node` method.